### PR TITLE
Add Edit tenant API (Lombiq Technologies: OCORE-117)

### LINKED ISSUE #13110: `Update()` endpoint for the Tenant's ApiController
### Is your feature request related to a problem? Please describe.
This is kind of related to #13085. Currently, there is no way to update tenant settings from API calls.

### Describe the solution you'd like
There could be an `Update()` endpoint in the Tenant's ApiController, which takes care of this and enable us to modify the tenant settings either if the tenant is uninitialized or running. The updateable properties should be the same as what is displayed on the tenant editor on the Admin UI.

### PR DESCRIPTION
**Pull request by hishamco**
Fixes #13110

### CODE REVIEW

**Comment by hishamco** [comment]
@MZole while the API is not managed from CMS, is there any other configuration that we need to add, such as recipes, feature profiles?

**Comment by hishamco** [comment]
@Piedone just one thing before adding a unit test for this - if it's needed - can we use `TenantViewModel` instead for both create and edit tenant? Another think I dislike the term `ViewModel` especially if working with APIs coz no views involved, can we use `Model` instead?

**Comment by Piedone** [comment]
If you don't see any differences between the two usages that make this awkward then sure.

Yeah, the API shouldn't really have a view model. If we want to be purists, the API and the MVC controller should use different model classes, so you can do that (with a common base class, perhaps, to make the validator work).

**Comment by hishamco** [comment]
Or let us introduce `TenentAPIModel` for APIs specific, I don't want to touch the base class coz `TenantViewModel` for MVC make sense

**Comment by Piedone** [comment]
Sounds good, but you'll need something common in the type hierarchy to be able reuse the tenant validator (an interface is suitable as well, of course). BTW the conventional naming would rather be `TenantApiModel`.

**Review by Piedone — CHANGES_REQUESTED** [review]

File: src/OrchardCore.Modules/OrchardCore.Tenants/Controllers/ApiController.cs [hunk 1]
```diff
@@ -24,6 +24,7 @@
 using OrchardCore.Mvc.Utilities;
 using OrchardCore.Recipes.Models;
 using OrchardCore.Setup.Services;
+using OrchardCore.Tenants.Models;
```
[the remark this answers is not in the document]
Comment (hishamco) [reply src/OrchardCore.Modules/OrchardCore.Tenants/Controllers/ApiController.cs:27]: May be this happen during merge conflict, coz I added two lines manually in the "Fork" git client

**Review by Piedone — APPROVED** [review]

**Comment by hishamco** [comment]
@MZole do you have any input on this, before we merge? This will be unique if I got two approves from two Zoltan :)

**Comment by MZole** [comment]
I have tested the edit endpoint with our ApiClient, and it works as intended, so I also approve! :smile:

File: src/OrchardCore.Modules/OrchardCore.Tenants/Controllers/ApiController.cs [hunk 2]
```diff
@@ -143,6 +144,51 @@ public async Task<IActionResult> Create(CreateApiViewModel model)
...
+            if (!await _authorizationService.AuthorizeAsync(User, Permissions.ManageTenants))
+            {
+                return this.ChallengeOrForbid("Api");
+            }
+
+            if (!_shellHost.TryGetSettings(model.Name, out var shellSettings))
+            {
+                return NotFound();
+            }
+
```
Comment (jtkech) [inline src/OrchardCore.Modules/OrchardCore.Tenants/Controllers/ApiController.cs:165]: @Piedone @hishamco 

What is missing here is the usage of `_tenantValidator.ValidateAsync()` that is used in the AdminController through `ValidateViewModelAsync(model, false)`.

Comment (Piedone) [reply src/OrchardCore.Modules/OrchardCore.Tenants/Controllers/ApiController.cs:165]: Right. Hisham, will you open a new PR?

Comment (hishamco) [reply src/OrchardCore.Modules/OrchardCore.Tenants/Controllers/ApiController.cs:165]: Fixed in #13160, also I did a little refactor to utilize the `ValidateModelAsync()` in both create and edit APIs

**Pull request merged**